File: app/src/main/java/ua/syt0r/furiganit/model/repository/hisotry/HistoryRepository.java
package ua.syt0r.furiganit.model.repository.hisotry;

import java.util.List;
import io.reactivex.Completable;
import io.reactivex.Single;
import ua.syt0r.furiganit.model.entity.HistoryItem;

public interface HistoryRepository {
    Completable add(HistoryItem historyItem);
    Completable remove(HistoryItem historyItem);
    Single<List<HistoryItem>> fetchHistory();
}

File: app/src/main/java/ua/syt0r/furiganit/model/repository/hisotry/HistoryRepository.kt
package ua.syt0r.furiganit.model.repository.hisotry

import io.reactivex.Completable
import io.reactivex.Single
import ua.syt0r.furiganit.model.entity.HistoryItem

interface HistoryRepository {
    fun add(historyItem: HistoryItem): Completable
    fun remove(historyItem: HistoryItem): Completable
    fun fetchHistory(): Single<List<HistoryItem>>
}
